### PR TITLE
Documentation initial commit

### PR DESCRIPTION
**Pull request by castroflavio**
Initial commit with documentation restructuration

### CODE REVIEW

File: docs/about/Changelog.md
Comment (JoaquinChung) [inline docs/about/Changelog.md:1]: Minor typo fixes:
Second bullet under [Added], maybe change Haproxy to HAProxy and capitalize Deleted

File: docs/dev-guide/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Developer Guide
+
+## Software components
+* **SciStream Data Server (S2DS):** software that runs on gateway nodes. It acts as a buffer-and-forward agent.
+* **SciStream User Client (S2UC):** software that the end user and/or workflow engines/tools acting on behalf of the user interact with and provide relevant information (e.g., ID of a HPC job, ID of an experiment or data acquisition job on a scientific instrument, shared secret for secure communication with the user job (application) at the producer and consumer) to orchestrate end-to-end data streaming.
+* **SciStream Control Server (S2CS):** a software running on one of the gateway nodes. It interacts with S2UC, data producer/consumer and S2DS.
+
+## S2DS and ProxyContainer Documentation
+
+The code consists mainly of the `S2DS` class and `ProxyContainer` subclasses, including `Haproxy`, `Nginx`, `Janus`, and `DockerSock`, which serve to establish and manage the data streaming pipelines using various networking and container technologies.
```
Comment (JoaquinChung) [inline docs/dev-guide/README.md:10]: Haproxy ---> HAProxy

File: docs/dev-guide/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+### ProxyContainer and Subclasses
+
+The `ProxyContainer` class and its subclasses (`Haproxy`, `Nginx`, `Janus`, `DockerSock`) are a specific implementation of S2DS and they abstract the complexities of deploying proxy containers to facilitate. These classes are designed to work with different Docker plugins and configurations, allowing for flexible deployment scenarios. Each subclass specifies its own container configuration, including the image name, container name, and configuration file locations. The `start` method in each class leverages Docker APIs to deploy and manage containers based on the provided configuration. Additionally, the `update_listeners` method in `ProxyContainer` demonstrates how to dynamically update listener configurations using Jinja2 templates.
```
Comment (JoaquinChung) [inline docs/dev-guide/README.md:18]: First sentence sounds incomplete, unless facilitate should be changed to facilities?

File: docs/figures/auth.rest [hunk 3]
```diff
@@ -0,0 +1,44 @@
+======================================
+Scistream Client Authentication Workflow
+======================================
+
+Making a request without Local Credentials
+------------------------------------------
+
+**Case A: Scistream Server Does Not Require Authentication (Default Case)**
+
+- *Server Command*::
```
Comment (JoaquinChung) [inline docs/figures/auth.rest:10]: Should we change this to *Command for initializing server:*?

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      ...               # other files
+    tests/
+      test_s2cs.py      # pytest tests
+      ...               # other tests
+
+### Service
+      The protocol should enable high-speed, memory-to-memory data streaming in scientific environments
+      by establishing streaming data channels between two remote facilities using our reference architecture:
+
+      ![alt text](figures/simple-arch.png "SciStream architecture")
```
Comment (JoaquinChung) [inline docs/index.md:44]: Call to the figures may not be rendering well. 
[scistream-simple-arch-with-software.pdf](https://github.com/scistream/scistream-proto/files/14317047/scistream-simple-arch-with-software.pdf)
Maybe we should use this newer image instead.

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      * **SciStream Data Server (S2DS):** software that runs on gateway nodes. It acts as a buffer-and-forward agent.
+      * **SciStream User Client (S2UC):** software that the end user and/or workflow engines/tools acting on behalf of the user interact with and provide relevant information (e.g., ID of a HPC job, ID of an experiment or data acquisition job on a scientific instrument, shared secret for secure communication with the user job (application) at the producer and consumer) to orchestrate end-to-end data streaming.
+      * **SciStream Control Server (S2CS):** a software running on one of the gateway nodes. It interacts with S2UC, data producer/consumer and S2DS.
+
+### Environment
+      * S2UC communicates with producer/consumer S2CS over a private LAN/WAN or the Internet
+      * S2CS and S2DS communicate over a LAN
+      * Messages can be lost or corrupted
+
+### Vocabulary of Messages
```
Comment (JoaquinChung) [inline docs/index.md:58]: Probably don't need this (old stuff)

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+      * **Requests:** { REQ, ReqListeners }
+      * **Responses:** { RESP, ProdLstn }
+      * **Commands:** { StartLstn, Hello, UpdateTargets, StartConn, Connect, REL }
+
+### Procedure Rules (Informal)
```
Comment (JoaquinChung) [inline docs/index.md:63]: This is probably outdated too

File: docs/quickstart.md [hunk 7]
```diff
@@ -131,46 +140,29 @@ Make sure that you start the poetry environment with the correct python version
...
-### Collaboration Diagram
-
-![alt text](figures/collaboration-diagram.png "SciStream collaboration diagram")
-
-### Sequence Diagram
-
-![alt text](figures/scistream-protocol-simple.png "SciStream sequence diagram")
+## Authentication
+
+How to create scopes:
```
Comment (JoaquinChung) [inline docs/quickstart.md:145]: Need to add code formatting to everything after this line

**Review by JoaquinChung — APPROVED** [review]
In the documentation files, there are only minor typographical suggestions. This can be taken care of later.

File: docs/dev-guide/README.md (hunk 2 above)
Comment (castroflavio) [reply docs/dev-guide/README.md]: I'm no longer using this page

File: docs/figures/auth.rest (hunk 3 above)
Comment (castroflavio) [reply docs/figures/auth.rest]: I also modified this, so this comment is now outdated

File: docs/dev-guide/README.md (hunk 1 above)
Comment (castroflavio) [reply docs/dev-guide/README.md]: I also modified this, so this comment is now outdated

File: docs/about/Changelog.md
Comment (castroflavio) [reply docs/about/Changelog.md:1]: Well, the name of the python class is "Haproxy", so the documentation is technically consistent with itself.

File: docs/index.md (hunk 6 above)
Comment (castroflavio) [reply docs/index.md]: Yeah I haven't deleted things yet. We can do that later.

**Pull request merged**